Move the list of coercions to the reference (#290)
rust-lang/nomicon: src/coercions.md

# Coercions

Types can implicitly be coerced to change in certain contexts.
These changes are generally just *weakening* of types, largely focused around pointers and lifetimes.
They mostly exist to make Rust "just work" in more cases, and are largely harmless.

For an exhaustive list of all the types of coercions, see the [Coercion types] section on the reference.

> ## Cases where coercions do not perform

Reviewer: ```suggestion
## Cases where coercions do not happen
```

(but I would drop the subheading altogether -- this is already a very short page now)

> Note that we do not perform coercions when matching traits (except for receivers, see below).

Reviewer: ```suggestion
Note that we do not perform coercions when matching traits (except for receivers, see the next page).
```

This is a per-existing error, but still should be fixed.

}
```

> which fails like the following:

Reviewer: ```suggestion
which fails as follows:
```
